### ISSUE TITLE
redirection_capability_check is not restricting access properly

### ISSUE THREAD

**Issue by korynorthrop** [johngodley/redirection#2262]
### Summary

I am trying to allow Editors access to view and create redirects as explained [here](https://redirection.me/developer/permissions/), but restrict their access everywhere else.

### Actual Behavior

Using an Editor account, I was able to access the plugin settings area and was able change Options, which shouldn't be possible if I understand what these Permissions filters are supposed to do. In fact, on my production site, which I haven't modified yet, I am finding that Editors can already access and modify all areas of the Redirection plugin settings.

### Steps to reproduce
I added the following to my functions.php file: 
```
add_filter( 'redirection_role', function( $role ) {
  return 'edit_posts';  // Add your chosen capability or role here
} );
add_filter( 'redirection_capability_check', function( $capability, $permission_name ) {
  if ( $permission_name === 'redirection_cap_redirect_manage' || $permission_name === 'redirection_cap_redirect_add' ) {
    return $capability;
  }
  return 'manage_options';
}, 10, 2 );
```

Then logged into my site as a User with Editor permissions then I went to Tools > Redirection > Options to try to modify some of the plugin options, which I was able to do.

### Environment

- Firefox 75.0 and Chrome 79
- Wordpress 5.3.2
- Redirection 4.7.1

**Comment by johngodley** [comment]
> I am finding that Editors can already access and modify all areas of the Redirection plugin settings

I am unable to reproduce this with an unmodified version of Redirection.

> Then logged into my site as a User with Editor permissions then I went to Tools > Redirection > Options to try to modify some of the plugin options, which I was able to do.

I was also unable to do this. Using the exact same code you posted above my Redirection looks like this:

![image](https://user-images.githubusercontent.com/1277682/81403665-d4523500-912b-11ea-9498-aeb9c0ea8345.png)

As you can there is no 'options' menu, and if I try and access it directly I am redirected back to the page I can access.

**Comment by korynorthrop** [comment]
Thanks for your reply. Perhaps I had some lingering capabilities that were added to our Editor role. I installed the User Role Editor plugin to reset the capabilities for the Editor and found that the 'options' menu was still showing up. However, when I enabled the aforementioned filters and change the capability to an Admin-only capability it worked as expected. So, probably just something residual from previous permissions work that was done on the site. Sorry to ring a false alarm.

```
add_filter( 'redirection_capability_check', function( $capability, $permission_name ) {
  if ( $permission_name === 'redirection_cap_redirect_manage' || $permission_name === 'redirection_cap_redirect_add' ) {
    return $capability;
  }
  return 'delete_users'; // admin-level capability
}, 10, 2 );
```
